recommendations/index.md

```diff
@@ -149,3 +149,53 @@ Although JSON API does not specify the format of date and time fields, it is
 recommended that servers align with ISO 8601. [This W3C
 NOTE](http://www.w3.org/TR/NOTE-datetime) provides an overview of the
 recommended formats.
+=======
+
+## Asynchronous Processing <a href="#asynchronous-processing" id="asynchronous-processing" class="headerlink"></a>
+
+Consider a situation when you need to create a resource and the operation takes long time to complete.
+
+The request **SHOULD** return a status `202 Accepted` with a link in Header/Location.
+
+```text
+HTTP/1.1 202 Accepted
+Location: https://example.com/photos/queue/5234
+Content-Type: application/vnd.api+json
+```
+
+The location gave **SHOULD** return `200 Ok` with the status of the selected job. The body **MAY** contains `"actions"` to manage the job.
+
+```json
+{
+  "links": {
+    "self": "/photos/queue/5234"
+  },
+    "data": {
+      "type": "queue job",
+      "id": "5234",
+      "status": "Process failed, see errors",
```

Comment:
`"status"` and `"errors"` need to be in an `"attributes"` bucket to comply with JSON API.